docsource/functionalRequirements.rst

```diff
@@ -1,2 +1,24 @@
 Functional Requirements
-========================
\ No newline at end of file
+========================
+
+1. Support the indexing of many data types (Genome, Assembly, Narrative etc). It is expected that there will be about 100 types
+
+2. Support the indexing of versions of types. It is expected that each object can have on average about 100 versions.
+
+3. Support the indexing of instance versions. A specific version of a specific object may have mutable data that me require index updates.
+
+4. In general flatten or denormalize data to improve search performance. But, where applicable, use the parent-child relationship to minimize load on indexing when parent data changes and this change can affect upwards of thousands of children.
```

Comment:
The second sentence seems like an implementation detail rather than a function requirement. The functional requirement sounds like 'properly balance search performance and update times for mutable data' or something like that, but that's a bit too vague.

Reply:
Changed.